packages/core/services/maps-api-loader/lazy-maps-api-loader.ts

```diff
@@ -87,14 +88,24 @@ export class LazyMapsAPILoader extends MapsAPILoader {
   protected readonly _SCRIPT_ID: string = 'agmGoogleMapsApiScript';
   protected readonly callbackName: string = `agmLazyMapsAPILoader`;
 
-  constructor(@Optional() @Inject(LAZY_MAPS_API_CONFIG) config: any = null, w: WindowRef, d: DocumentRef) {
+  constructor(
+    @Optional() @Inject(LAZY_MAPS_API_CONFIG) config: any = null,
+    w: WindowRef,
+    d: DocumentRef,
+    @Inject(PLATFORM_ID) private platformId: Object,
+  ) {
     super();
     this._config = config || {};
     this._windowRef = w;
     this._documentRef = d;
   }
 
   load(): Promise<void> {
+    if (!isPlatformBrowser(this.platformId)) {
+      // The code is running on the server, skip
+      return Promise.resolve();
```

Comment:
When someone uses the loader, this would crash on the server side (because the code will assume that the google maps api is loaded) - I think we don't need a change here and a notice in a "running AGM with in universal" apps would be better

Reply:
Another note: we need this change in the agm-map component (this would also prevent the loading in most of the use cases)

Reply:
Good point, moving the logic to the map component is much cleaner